File: pgm/ABC_Kotlin/ABC_299_A.kt
// import java.io.BufferedOutputStream
// import java.io.BufferedReader
// import java.io.DataInputStream
// import java.io.FileInputStream
// import java.io.IOException
// import java.io.InputStreamReader
// import java.io.PrintWriter
// import java.lang.StringBuilder
// import java.time.*
// import java.util.Scanner
// import java.util.StringTokenizer

// import kotlin.math.*

const val dot: String = "."
const val bar: String = "|"
const val ast: String = "*"
const val strIn: String = "in"
const val strOut: String = "out"

fun main() {
    readln()
    val S: String = readln()
    val escBar: String = Regex.escape(bar)
    val escDot: String = Regex.escape(dot)
    val escAst: String = Regex.escape(ast)
    val pattern: String = escBar.plus(escDot).plus(ast).plus(escAst).plus(escDot).plus(ast).plus(escBar)
    if (Regex(pattern).containsMatchIn(S)) {
        println(strIn)
    } else {
        println(strOut)
    }
    kotlin.system.exitProcess(0)
}

File: pgm/ABC_Dart/ABC_299_A.dart
// import 'dart:collection';
import 'dart:io';
// import 'dart:math';
// import 'dart:typed_data';

void main() {
  stdin.readLineSync()!;
  final String S = stdin.readLineSync()!;
  const String dot = ".";
  const String bar = "|";
  const String ast = "*";
  const String strIn = "in";
  const String strOut = "out";
  final String escBar = RegExp.escape(bar);
  final String escAst = RegExp.escape(ast);
  final String pattern = "$escBar$dot$ast$escAst$dot$ast$escBar";
  if (RegExp(pattern).hasMatch(S)) {
    print(strIn);
  } else {
    print(strOut);
  }
  exit(0);
}
